packages/synapse-react-client/src/components/dataaccess/RejectDataAccessRequestModal.test.tsx

```diff
@@ -0,0 +1,108 @@
+import { SubmissionState } from '@sage-bionetworks/synapse-types'
+import { act, render, waitFor } from '@testing-library/react'
+import failOnConsoleError from 'jest-fail-on-console'
+import React from 'react'
+import { mockSubmittedSubmission } from '../../mocks/dataaccess/MockSubmission'
+import { MOCK_ACCESS_TOKEN } from '../../mocks/MockSynapseContext'
+import { rest, server } from '../../mocks/msw/server'
+import SynapseClient from '../../synapse-client'
+import { createWrapper } from '../../testutils/TestingLibraryUtils'
+import { DATA_ACCESS_SUBMISSION_BY_ID } from '../../utils/APIConstants'
+import {
+  BackendDestinationEnum,
+  getEndpoint,
+} from '../../utils/functions/getEndpoint'
+import { REJECT_SUBMISSION_CANNED_RESPONSES_TABLE } from '../../utils/SynapseConstants'
+import { CannedRejectionDialog } from '../CannedRejectionDialog/CannedRejectionDialog'
+import RejectDataAccessRequestModal, {
+  DEFAULT_MESSAGE_APPEND,
+  DEFAULT_MESSAGE_PREPEND,
+  RejectDataAccessRequestModalProps,
+  REJECTION_FORM_PROMPT_COPY,
+} from './RejectDataAccessRequestModal'
+
+// Mock the CannedRejectionDialog component
+jest.mock('../CannedRejectionDialog/CannedRejectionDialog', () => ({
+  CannedRejectionDialog: jest.fn().mockImplementation(() => <div />),
+}))
+
+const mockCannedRejectionDialog = jest.mocked(CannedRejectionDialog)
+
+const props: RejectDataAccessRequestModalProps = {
+  submissionId: mockSubmittedSubmission.id,
+  tableId: REJECT_SUBMISSION_CANNED_RESPONSES_TABLE,
+  open: true,
+  onClose: jest.fn(),
+}
+
+const updateSubmissionStatusSpy = jest.spyOn(
+  SynapseClient,
+  'updateSubmissionStatus',
+)
+
+function renderComponent() {
+  return render(<RejectDataAccessRequestModal {...props} />, {
+    wrapper: createWrapper(),
+  })
+}
+
+describe('RejectDataAccessRequestModal', () => {
```

Comment:
For `RejectDataAccessRequestModal`, rewrite tests to mock the `CannedRejectionDialog` and just verify the callback works.